### ISSUE TITLE
useQueries tries to inject default query client when query array is empty

### ISSUE THREAD

**Issue by stellR42** [TanStack/query#4615]
### Describe the bug

I'm using `vue-query` in a packaged library so instead of using the `VueQueryPlugin` I define my own query client and pass it to every query call. The issue is that when using `useQueries`, when the query array is empty I get this error: ` No 'queryClient' found in Vue context, use 'VueQueryPlugin' to properly initialize the library.`

This seems to be because `vue-query` can't find a passed-in query client (obviously, since there are no queries) so it defaults to trying to inject the default query client that `VueQueryPlugin` would have created.

### Your minimal, reproducible example

https://codesandbox.io/s/angry-estrela-jsovt5?file=/src/App.vue

### Steps to reproduce

Error is displayed as soon as the sandbox loads.

### Expected behavior

It should bail out as soon as it sees that the query array is empty instead of making assumptions about what the defaults should be. Or maybe there can be an option to pass a query client to the `useQueries` function without relying on the query client in each query config.

### How often does this bug happen?

Every time

### Platform

Windows Chrome v102

### Tanstack Query version

   @tanstack/vue-query 4.19.1

**Comment by TkDodo** [comment]
good point. I think the api for passing in a custom queryClient should actually be different for useQueries. In the react adapter, we have a similar thing with passing in a custom react context (this will change in v5 when we will _also_ pass in a custom queryClient).

The api we have in react is:

```
useQueries({ queries: [ query1, query2, ... queryN ], context })
```

where `query1` to `queryN` is: `Omit<UseQueryOptions<TQueryFnData, TError, TData, TQueryKey>, 'context'>` so that you cannot pass in a context for each query, but instead, have to provide one at the very top for all queries.

Yes, this limits all queries to use the same context / queryClient, but I think this is alright.

As a non-breaking fix, we could allow both - `queryClient` at the top as well as `queryClient` on each query, but discourage the latter (with a warning), and remove it in v5. Thoughts @DamianOsipiuk ?

**Comment by DamianOsipiuk** [comment]
Yup, definitely we should provide `queryClient` instance next-to `queries` , not individually.
